### DISCUSSION TITLE
Generate A/B labels for a bunch of hex IDs

### DISCUSSION THREAD

**Discussion by bkowshik** [uber/h3#440] [Q&A] [answered]
I want to label a bunch of hex IDs into A/B so that With the code snippet below, I randomly assign a A/B label to hex IDs and the output looks like the map below. Want to know what are other strategies to label hex IDs as A/B.

```python
import random

def split_into_ab(hex_id):
    return random.randint(1, 2) == 1
```

<img width="1680" alt="Screenshot 2021-03-27 at 10 54 05 AM" src="https://user-images.githubusercontent.com/2899501/112710951-d6131b80-8eea-11eb-82a8-d733cf4cd3a6.png">

**Comment by dfellis** [comment, marked as answer]
If you want it based on the `hex_id` and therefore the same every time, what about:

```py
def split_into_ab(hex_id):
    return int(hex_id, 16) % 2 == 1
```

**Reply by bkowshik** [reply]
Thank you @dfellis 

Not sure if this is because of the resolution of the hex IDs or the area I am working with. All the hex IDs in my list, end with an odd number and thus `% 2 = 1` for all the hex IDs. Some sample hex IDs from the list of hex IDs that formed the map above are below for reference.

```
hex_id          ,integer_value
8860145101fffff ,614179795317555199
8860145103fffff ,614179795319652351
8860145105fffff ,614179795321749503
8860145107fffff ,614179795323846655
8860145109fffff ,614179795325943807
886014510bfffff ,614179795328040959
886014510dfffff ,614179795330138111
8860145111fffff ,614179795334332415
8860145113fffff ,614179795336429567
```

**Reply by dfellis** [reply]
Oh,  right, the bit format is not random enough to work it that way. If you integer divide it by a prime number before doing the modulus that should be enough for your purposes, or you could feed it into a non-cryptographic hash function like [xxHash](https://cyan4973.github.io/xxHash/) though that is decidedly overkill here.

**Reply by dfellis** [reply]
[Integer division in Python 3](https://pythonexamples.org/python-division/) requires the `//` symbol, so the modified version would be:

```py
def split_into_ab(hex_id):
    return int(hex_id, 16) // 3 % 2
```

Using `3` here as the prime number since there are only two buckets you're going into, which should keep neighbors from being too regular with each other, I think? It's been a while since I've read up on hashing.

**Reply by dfellis** [reply]
Just tested in a REPL, I think that version is nicely mixed:

![Screenshot from 2021-03-26 23-57-12](https://user-images.githubusercontent.com/765551/112712825-088f2b00-8e8f-11eb-9341-aa877ea903bb.png)
